File: 1° Período - Introdução a Programação 1/Exercicios em Python/Lista 5 - Python/q5.py
#Escreva um algoritmo que leia um número e diga se este número é perfeito ou não. Um número
#perfeito é aquele que é igual à soma de seus divisores, excluindo o próprio número. Exemplos:
#1 + 2 + 3 = 6;
#1 + 2 + 4 + 7 = 28.

x=int(input("N: "))
i=1
soma=0
while i<x:
    if x%i==0:
        soma=soma+i
        print(i)
    i=i+1
print("Soma dos divisores: ", soma)
if soma==x:
    print("PERFEITO")
else:
    print("NÃO É PERFEITO")
    

File: 1° Período - Introdução a Programação 1/Exercicios em C/Lista 5 - C/q5.c
//. Escreva um algoritmo que leia um número e diga se este número é perfeito ou não. Um número
//perfeito é aquele que é igual à soma de seus divisores, excluindo o próprio número. Exemplos:
//1 + 2 + 3 = 6;
//1 + 2 + 4 + 7 = 28.
#include<stdio.h>
int main(){
	int x, i, soma;
	printf("Insira um numero: ");
	scanf("%d", &x);
	i=1;
	soma=0;
	while(i<x){
		if(x%i==0){
			soma=soma+i;
	        printf("%d\n", i);
		}
		i=i+1;       
	}
	printf("\nSoma dos divisores: %d", soma);
	if(soma==x){
		printf("\nPERFEITO");
	}
	else{
		printf("\nNÃO É PERFEITO");	
	}	    
}
